### PR TITLE
Rename `Check` to `Subject`

### PR DESCRIPTION
**Pull request by natebosch**
The word "check" is a bit overused in the documentation, and it's a
little confusing to use it as a noun. "Subject" is unambiguously a noun,
and it makes the docs read nicer in some places.

### CODE REVIEW

**Comment by natebosch** [comment]
I also considered `Target`, but I think I prefer `Subject`. I'm open to opinions and other suggestions.

File: pkgs/checks/README.md [hunk 1]
```diff
@@ -44,6 +45,6 @@ checkThat(actual).deepEquals(expected);
 
 ## Differences in behavior from matcher
 
-- The `equals` Matcher performed a deep equality check on collections.
-  `.equals()` check will only correspond to [operator ==] so some tests may need
-  to replace `.equals()` with `.deepEquals()`. **TODO: implement `deepEquals`**
+-   The `equals` Matcher performed a deep equality check on collections.
```
Comment (kevmoo) [inline pkgs/checks/README.md:48]: Indent diff intentional here?

File: pkgs/checks/lib/src/extensions/async.dart [hunk 2]
```diff
@@ -78,7 +78,7 @@ extension FutureChecks<T> on Check<Future<T>> {
 ///
 /// Streams should be wrapped in user test code so that any reuse of the same
 /// Stream, and the full stream lifecycle, is explicit.
-extension StreamChecks<T> on Check<StreamQueue<T>> {
+extension StreamChecks<T> on Subject<StreamQueue<T>> {
```
Comment (kevmoo) [inline pkgs/checks/lib/src/extensions/async.dart:81]: ditto

File: pkgs/checks/lib/src/extensions/function.dart [hunk 3]
```diff
@@ -4,19 +4,19 @@
 
 import 'package:checks/context.dart';
 
-extension ThrowsCheck<T> on Check<T Function()> {
+extension ThrowsCheck<T> on Subject<T Function()> {
```
Comment (kevmoo) [inline pkgs/checks/lib/src/extensions/function.dart:7]: dinno

**Review by kevmoo — COMMENTED** [review]
LGTM – although we want to completely ditch `Check` from the package, right?

File: pkgs/checks/README.md (hunk 1 above)
Comment (natebosch) [reply pkgs/checks/README.md:48]: I think I probably autoformatted and this is what the internal formatter does.

**Review by jakemac53 — APPROVED** [review]

**Pull request merged**